--- a/lib/mcel.h
+++ b/lib/mcel.h
@@ -221,6 +221,11 @@
 
   /* An initial mbstate_t; initialization optimized for some platforms.
      For details about these and other platforms, see wchar.in.h.  */
+
+#if defined __UCLIBC__ && defined __GLIBC__
+#undef __GLIBC__
+#endif
+
 #if defined __GLIBC__ && 2 < __GLIBC__ + (2 <= __GLIBC_MINOR__)
   /* Although only a trivial optimization, it's worth it for GNU.  */
   mbstate_t mbs; mbs.__count = 0;
