diff -Naur a/po/Makefile.in.in b/po/Makefile.in.in
--- a/po/Makefile.in.in	2013-05-07 17:06:36.000000000 +0400
+++ b/po/Makefile.in.in	2017-01-27 12:38:41.602281100 +0300
@@ -8,8 +8,8 @@
 # Please note that the actual code of GNU gettext is covered by the GNU
 # General Public License and is *not* in the public domain.
 #
-# Origin: gettext-0.18
-GETTEXT_MACRO_VERSION = 0.18
+# Origin: gettext-0.19
+GETTEXT_MACRO_VERSION = 0.19
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
