--- a/configure
+++ b/configure
@@ -13984,66 +13984,6 @@ fi
 
 
 
-# Check for bfd.h (to have proper backtraces without using wxExecute)
-
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for bfd" >&5
-$as_echo_n "checking for bfd... " >&6; }
-	result=no
-	for bfd_ldadd in "" "${LIBINTL}" "-ldl" "-ldl ${LIBINTL}"; do
-		mule_backup_LIBS="$LIBS"
-		mule_backup_LDFLAGS="$LDFLAGS"
-		LIBS="-lbfd -liberty ${bfd_ldadd} ${ZLIB_LIBS} $LIBS"
-		LDFLAGS="$LDFLAGS ${ZLIB_LDFLAGS}"
-		cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-
-				#include <ansidecl.h>
-				#include <bfd.h>
-
-int
-main ()
-{
-
-				char *dummy = bfd_errmsg(bfd_get_error());
-
-  ;
-  return 0;
-}
-
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-
-			result=yes
-			BFD_CPPFLAGS="-DHAVE_BFD"
-			BFD_LIBS="-lbfd -liberty ${bfd_ldadd}"
-			LIBS="$mule_backup_LIBS"
-			LDFLAGS="$mule_backup_LDFLAGS"
-			break
-
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-		LIBS="$mule_backup_LIBS"
-		LDFLAGS="$mule_backup_LDFLAGS"
-	done
-
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $result" >&5
-$as_echo "$result" >&6; }
-
-	if test $result = no; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING:
-        bfd.h not found or unusable, please install binutils development
-        package if you are a developer or want to help testing aMule" >&5
-$as_echo "$as_me: WARNING:
-        bfd.h not found or unusable, please install binutils development
-        package if you are a developer or want to help testing aMule" >&2;}
-_mule_warning_10=yes
-_mule_has_warnings=yes
-
-fi
-
-
 
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for <execinfo.h> and backtrace()" >&5
 $as_echo_n "checking for <execinfo.h> and backtrace()... " >&6; }
