--- a/Makefile.am	2020-10-13 08:53:51.000000000 +0300
+++ b/Makefile.am	2020-11-23 15:21:49.998516400 +0300
@@ -195,7 +195,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+DIST_SUBDIRS = $(SUBDIRS) packages include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -306,8 +306,8 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
+	#(cd docs && $(MAKE) install)
+	#(cd docs/libcurl && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
