--- a/lib/vtls/mbedtls.c	2019-03-25 11:42:50.000000000 +0300
+++ b/lib/vtls/mbedtls.c	2019-04-04 12:09:53.600586600 +0300
@@ -818,7 +818,7 @@
 {
 #ifdef MBEDTLS_VERSION_C
   /* if mbedtls_version_get_number() is available it is better */
-  unsigned int version = mbedtls_version_get_number();
+  unsigned int version = MBEDTLS_VERSION_NUMBER;
   return msnprintf(buffer, size, "mbedTLS/%u.%u.%u", version>>24,
                    (version>>16)&0xff, (version>>8)&0xff);
 #else
