--- a/kernel/net/Kbuild
+++ b/kernel/net/Kbuild
@@ -1,5 +1,6 @@
 NOSTDINC_FLAGS += -I$(KDIR)/include
 EXTRA_CFLAGS := -DCONFIG_IP_SET_MAX=$(IP_SET_MAX)
+KBUILD_CFLAGS := $(filter-out -O%,$(KBUILD_CFLAGS)) -O3
 
 obj-m += netfilter/
 obj-m += sched/
--- a/kernel/net/netfilter/Kbuild
+++ b/kernel/net/netfilter/Kbuild
@@ -1,5 +1,6 @@
 NOSTDINC_FLAGS += -I$(KDIR)/include
 EXTRA_CFLAGS := -DCONFIG_IP_SET_MAX=$(IP_SET_MAX)
+KBUILD_CFLAGS := $(filter-out -O%,$(KBUILD_CFLAGS)) -O3
 
 obj-m += xt_set.o
 obj-m += ipset/
--- a/kernel/net/netfilter/ipset/Kbuild
+++ b/kernel/net/netfilter/ipset/Kbuild
@@ -1,5 +1,6 @@
 NOSTDINC_FLAGS += -I$(KDIR)/include
 EXTRA_CFLAGS := -DIP_SET_MAX=$(IP_SET_MAX)
+KBUILD_CFLAGS := $(filter-out -O%,$(KBUILD_CFLAGS)) -O3
 
 ip_set-y := ip_set_core.o ip_set_getport.o pfxlen.o
 obj-m += ip_set.o
--- a/kernel/net/sched/Kbuild
+++ b/kernel/net/sched/Kbuild
@@ -1,4 +1,5 @@
 NOSTDINC_FLAGS += -I$(KDIR)/include
 EXTRA_CFLAGS := -DCONFIG_IP_SET_MAX=$(IP_SET_MAX)
+KBUILD_CFLAGS := $(filter-out -O%,$(KBUILD_CFLAGS)) -O3
 
 obj-m += em_ipset.o
