--- a/Makefile.am	2021-02-09 13:11:25.000000000 +0300
+++ b/Makefile.am	2021-08-05 13:10:43.402800400 +0300
@@ -15,10 +15,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
-	include \
-	man-po \
-	po \
-	testsuite
+	include
 
 AM_CFLAGS = -Iproc
 LDADD = ./proc/libprocps.la $(CYGWINFLAGS)
--- a/configure.ac	2021-02-09 13:11:25.000000000 +0300
+++ b/configure.ac	2021-08-05 13:10:43.434072600 +0300
@@ -323,8 +323,5 @@
 
 AC_CONFIG_FILES([Makefile
                  include/Makefile
-                 man-po/Makefile
-                 po/Makefile.in
-                 proc/libprocps.pc
-                 testsuite/Makefile])
+                 proc/libprocps.pc])
 AC_OUTPUT
