--- a/config.c
+++ b/config.c
@@ -1557,7 +1557,7 @@
                             /* replace '~' with content of $HOME cause low-level functions
                              * like stat() do not support the glob ~
                              */
-                            const char *env_home = secure_getenv("HOME");
+                            const char *env_home = getenv("HOME");
                             char *new_key = NULL;
 
                             if (!env_home) {
