--- a/src/mkheader.c
+++ b/src/mkheader.c
@@ -82,9 +82,12 @@
     {"i486-pc-kfreebsd-gnu", "i686-pc-kfreebsd-gnu"},
     {"i586-pc-kfreebsd-gnu"},
 
-    {"x86_64-pc-linux-gnuhardened1", "x86_64-pc-linux-gnu" },
+    {"x86_64-openwrt-linux-gnu", "x86_64-pc-linux-gnu" },
 
     {"powerpc-unknown-linux-gnuspe", "powerpc-unknown-linux-gnu" },
+    {"arm-openwrt-linux-gnu", "arm-unknown-linux-gnueabi"},
+    {"mipsel-openwrt-linux-gnu", "mipsel-unknown-linux-gnu"},
+    {"mips-openwrt-linux-gnu", "mips-unknown-linux-gnu"},
 
     {"arm-unknown-linux-gnueabihf",  "arm-unknown-linux-gnueabi" },
     {"armv7-unknown-linux-gnueabihf"  },
