--- a/configure
+++ b/configure
@@ -5552,10 +5552,10 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking location of pid file" >&5
 $as_echo_n "checking location of pid file... " >&6; }
-if test -d /run ; then
-    PIDDIR=/run
-elif test -d /var/run ; then
+if test -d /var/run ; then
     PIDDIR=/var/run
+elif test -d /run ; then
+    PIDDIR=/run
 elif test -d /usr/run ; then
     PIDDIR=/usr/run
 else
--- a/configure.ac
+++ b/configure.ac
@@ -140,10 +140,10 @@
 AC_SUBST(systemdsystemunitdir, [$with_systemdsystemunitdir])
 
 AC_MSG_CHECKING(location of pid file)
-if test -d /run ; then
-    PIDDIR=/run
-elif test -d /var/run ; then
+if test -d /var/run ; then
     PIDDIR=/var/run
+elif test -d /run ; then
+    PIDDIR=/run
 elif test -d /usr/run ; then
     PIDDIR=/usr/run
 else
