--- a/blobmsg_json.c
+++ b/blobmsg_json.c
@@ -88,7 +88,7 @@
 	return ret;
 }
 
-bool blobmsg_add_json_from_file(struct blob_buf *b, const char *file)
+bool blobmsg_add_json_from_file(struct blob_buf *b, char *file)
 {
 	return __blobmsg_add_json(b, json_object_from_file(file));
 }
--- a/blobmsg_json.h
+++ b/blobmsg_json.h
@@ -28,7 +28,7 @@
 bool blobmsg_add_object(struct blob_buf *b, json_object *obj);
 bool blobmsg_add_json_element(struct blob_buf *b, const char *name, json_object *obj);
 bool blobmsg_add_json_from_string(struct blob_buf *b, const char *str);
-bool blobmsg_add_json_from_file(struct blob_buf *b, const char *file);
+bool blobmsg_add_json_from_file(struct blob_buf *b,  char *file);
 
 typedef const char *(*blobmsg_json_format_t)(void *priv, struct blob_attr *attr);
 
