--- jq-1.3/jv_print.c.orig	2014-03-15 04:38:21.009292257 +0800
+++ jq-1.3/jv_print.c	2014-03-15 04:17:57.179291404 +0800
@@ -49,7 +49,7 @@
   const char* cstart;
   int c = 0;
   char buf[32];
-  put_char('"', F, S);
+  //put_char('"', F, S);
   while ((i = jvp_utf8_next((cstart = i), end, &c))) {
     assert(c != -1);
     int unicode_escape = 0;
@@ -106,7 +106,7 @@
     }
   }
   assert(c != -1);
-  put_char('"', F, S);
+  //put_char('"', F, S);
 }
 
 enum { INDENT = 2 };
