--- a/src/jv_print.c
+++ b/src/jv_print.c
@@ -69,7 +69,9 @@
   const char* cstart;
   int c = 0;
   char buf[32];
+#if 0
   put_char('"', F, S, T);
+#endif
   while ((i = jvp_utf8_next((cstart = i), end, &c))) {
     assert(c != -1);
     int unicode_escape = 0;
@@ -126,7 +128,9 @@
     }
   }
   assert(c != -1);
+#if 0
   put_char('"', F, S, T);
+#endif
 }
 
 static void put_refcnt(struct dtoa_context* C, int refcnt, FILE *F, jv* S, int T){
