diff -Nuar a/configure b/configure
--- a/configure	2018-11-02 17:58:53.000000000 +0300
+++ b/configure	2018-12-11 18:22:46.279135400 +0300
@@ -15450,21 +15450,22 @@
 	LIBS="$LIB_gettimeofday $LIBS"
 fi
 
-ac_fn_c_check_member "$LINENO" "struct tm" "tm_gmtoff" "ac_cv_member_struct_tm_tm_gmtoff" "#include <time.h>
-"
-if test "x$ac_cv_member_struct_tm_tm_gmtoff" = xyes; then :
+# ac_fn_c_check_member "$LINENO" "struct tm" "tm_gmtoff" "ac_cv_member_struct_tm_tm_gmtoff" "#include <time.h>
+# "
+# if test "x$ac_cv_member_struct_tm_tm_gmtoff" = xyes; then :
 
 $as_echo "#define HAVE_TM_TM_GMT_OFF 1" >>confdefs.h
 
-fi
-
-ac_fn_c_check_member "$LINENO" "struct tm" "__tm_gmtoff" "ac_cv_member_struct_tm___tm_gmtoff" "#include <time.h>
-"
-if test "x$ac_cv_member_struct_tm___tm_gmtoff" = xyes; then :
-
-$as_echo "#define HAVE_TM___TM_GMT_OFF 1" >>confdefs.h
-
-fi
+# fi
+#
+#
+# ac_fn_c_check_member "$LINENO" "struct tm" "__tm_gmtoff" "ac_cv_member_struct_tm___tm_gmtoff" "#include <time.h>
+# "
+# if test "x$ac_cv_member_struct_tm___tm_gmtoff" = xyes; then :
+#
+# $as_echo "#define HAVE_TM___TM_GMT_OFF 1" >>confdefs.h
+#
+# fi
 
 
 # Check whether --enable-pthread-tls was given.
