--- a/sys-utils/unshare.c
+++ b/sys-utils/unshare.c
@@ -709,7 +709,9 @@
 		OPT_KILLCHILD,
 		OPT_KEEPCAPS,
 		OPT_MONOTONIC,
+#ifdef CLOCK_BOOTTIME
 		OPT_BOOTTIME,
+#endif
 		OPT_MAPUSER,
 		OPT_MAPUSERS,
 		OPT_MAPGROUP,
@@ -747,7 +749,9 @@
 		{ "root",	   required_argument, NULL, 'R'		    },
 		{ "wd",		   required_argument, NULL, 'w'		    },
 		{ "monotonic",     required_argument, NULL, OPT_MONOTONIC   },
+#ifdef CLOCK_BOOTTIME
 		{ "boottime",      required_argument, NULL, OPT_BOOTTIME    },
+#endif
 		{ NULL, 0, NULL, 0 }
 	};
 
@@ -775,7 +779,9 @@
 	time_t monotonic = 0;
 	time_t boottime = 0;
 	int force_monotonic = 0;
+#ifdef CLOCK_BOOTTIME
 	int force_boottime = 0;
+#endif
 
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
@@ -906,10 +912,12 @@
 			monotonic = strtoul_or_err(optarg, _("failed to parse monotonic offset"));
 			force_monotonic = 1;
 			break;
+#ifdef CLOCK_BOOTTIME
                 case OPT_BOOTTIME:
 			boottime = strtoul_or_err(optarg, _("failed to parse boottime offset"));
 			force_boottime = 1;
 			break;
+#endif
 
 		case 'h':
 			usage();
@@ -920,7 +928,11 @@
 		}
 	}
 
-	if ((force_monotonic || force_boottime) && !(unshare_flags & CLONE_NEWTIME))
+	if ((force_monotonic
+#ifdef CLOCK_BOOTTIME
+				|| force_boottime
+#endif
+				) && !(unshare_flags & CLONE_NEWTIME))
 		errx(EXIT_FAILURE, _("options --monotonic and --boottime require "
 			"unsharing of a time namespace (-t)"));
 
@@ -941,8 +953,10 @@
 	if (usermap || groupmap)
 		sync_with_child(pid_idmap, fd_idmap);
 
+#ifdef CLOCK_BOOTTIME
 	if (force_boottime)
 		settime(boottime, CLOCK_BOOTTIME);
+#endif
 
 	if (force_monotonic)
 		settime(monotonic, CLOCK_MONOTONIC);
