--- a/libmount/src/context_umount.c
+++ b/libmount/src/context_umount.c
@@ -24,6 +24,12 @@
 #include "strutils.h"
 #include "mountP.h"
 
+/* Define O_PATH if not defined */
+#ifndef O_PATH
+#define O_PATH		010000000
+#endif
+
+
 /*
  * umount2 flags
  */
--- a/sys-utils/fstrim.c
+++ b/sys-utils/fstrim.c
@@ -50,6 +50,10 @@
 
 #include <libmount.h>
 
+/* Define O_PATH if not defined */
+#ifndef O_PATH
+#define O_PATH		010000000
+#endif
 
 #ifndef FITRIM
 struct fstrim_range {
