--- a/libmount/src/context.c
+++ b/libmount/src/context.c
@@ -45,6 +45,7 @@
 #include <sys/wait.h>
 
 #include "strv.h"
+#include "namespace.h"
 
 /**
  * mnt_new_context:
--- a/libmount/src/hook_subdir.c
+++ b/libmount/src/hook_subdir.c
@@ -37,6 +37,7 @@
 
 #include "mountP.h"
 #include "fileutils.h"
+#include "namespace.h"
 #include "linux_version.h"
 
 struct hookset_data {
--- a/lsfd-cmd/lsfd.c
+++ b/lsfd-cmd/lsfd.c
@@ -58,6 +58,7 @@
 #include "pidfd-utils.h"
 
 #include "lsfd.h"
+#include "namespace.h"
 
 /* Make sure this ifdef block comes after all the includes since
  * c.h is required for the case where the system does not have kcmp.h */
--- a/lsfd-cmd/sock-xinfo.c
+++ b/lsfd-cmd/sock-xinfo.c
@@ -49,6 +49,7 @@
 #include "lsfd.h"
 #include "pidfd-utils.h"
 #include "sock.h"
+#include "namespace.h"
 
 static void load_xinfo_from_proc_icmp(ino_t netns_inode, enum sysfs_byteorder byteorder);
 static void load_xinfo_from_proc_icmp6(ino_t netns_inode, enum sysfs_byteorder byteorder);
--- a/sys-utils/nsenter.c
+++ b/sys-utils/nsenter.c
@@ -55,6 +55,7 @@
 #include "pathnames.h"
 #include "pidfd-utils.h"
 #include "linux_version.h"
+#include "namespace.h"
 #include "pidutils.h"
 
 #if defined(HAVE_STRUCT_NSFS_FILE_HANDLE) && defined(HAVE_PIDFD_OPEN)
--- a/tests/helpers/test_mkfds.c
+++ b/tests/helpers/test_mkfds.c
@@ -72,6 +72,7 @@
 #include <sys/wait.h>
 #include <time.h>
 #include <unistd.h>
+#include "namespace.h"
 
 #define EXIT_EPERM  18
 #define EXIT_ENOPROTOOPT 19
