--- a/libmount/src/context.c
+++ b/libmount/src/context.c
@@ -46,6 +46,7 @@
 
 #include "mount-api-utils.h"
 #include "strv.h"
+#include "namespace.h"
 
 /**
  * mnt_new_context:
--- a/libmount/src/hook_subdir.c
+++ b/libmount/src/hook_subdir.c
@@ -19,6 +19,7 @@
 
 #include "mountP.h"
 #include "fileutils.h"
+#include "namespace.h"
 
 struct hookset_data {
 	char *subdir;
--- a/lsfd-cmd/lsfd.c
+++ b/lsfd-cmd/lsfd.c
@@ -57,6 +57,7 @@
 #include "pathnames.h"
 
 #include "lsfd.h"
+#include "namespace.h"
 
 /* Make sure this ifdef block comes after all the includes since
  * c.h is required for the case where the system does not have kcmp.h */
--- a/lsfd-cmd/sock-xinfo.c
+++ b/lsfd-cmd/sock-xinfo.c
@@ -48,6 +48,7 @@
 #include "lsfd.h"
 #include "pidfd-utils.h"
 #include "sock.h"
+#include "namespace.h"
 
 static void load_xinfo_from_proc_icmp(ino_t netns_inode, enum sysfs_byteorder byteorder);
 static void load_xinfo_from_proc_icmp6(ino_t netns_inode, enum sysfs_byteorder byteorder);
--- a/sys-utils/nsenter.c
+++ b/sys-utils/nsenter.c
@@ -52,6 +52,7 @@
 #include "pathnames.h"
 #include "pidfd-utils.h"
 #include "linux_version.h"
+#include "namespace.h"
 
 static struct namespace_file {
 	int nstype;
--- a/tests/helpers/test_mkfds.c
+++ b/tests/helpers/test_mkfds.c
@@ -71,6 +71,7 @@
 #include <sys/wait.h>
 #include <time.h>
 #include <unistd.h>
+#include "namespace.h"
 
 #define EXIT_EPERM  18
 #define EXIT_ENOPROTOOPT 19
