diff -Nur avahi-0.6.31/avahi-core/querier-test.c avahi-0.6.31/avahi-core/querier-test.c
--- avahi-0.6.31/avahi-core/querier-test.c	2010-08-26 08:51:38.990153000 +0800
+++ avahi-0.6.31/avahi-core/querier-test.c	2017-01-05 22:40:16.582172633 +0800
@@ -35,7 +35,7 @@
 #include <avahi-core/lookup.h>
 
 #define DOMAIN NULL
-#define SERVICE_TYPE "_http._tcp"
+#define SERVICE_TYPE "_mywifiext._tcp"
 
 static AvahiSServiceBrowser *service_browser1 = NULL, *service_browser2 = NULL;
 static const AvahiPoll * poll_api = NULL;
@@ -106,9 +106,9 @@
     service_browser1 = avahi_s_service_browser_new(server, AVAHI_IF_UNSPEC, AVAHI_PROTO_UNSPEC, SERVICE_TYPE, DOMAIN, 0, sb_callback, NULL);
     assert(service_browser1);
 
-    poll_api->timeout_new(poll_api, avahi_elapse_time(&tv, 10000, 0), create_second_service_browser, NULL);
+    //poll_api->timeout_new(poll_api, avahi_elapse_time(&tv, 10000, 0), create_second_service_browser, NULL);
 
-    poll_api->timeout_new(poll_api, avahi_elapse_time(&tv, 60000, 0), quit, NULL);
+    poll_api->timeout_new(poll_api, avahi_elapse_time(&tv, 10000, 0), quit, NULL);
 
 
     for (;;)
diff -Nur avahi-0.6.31/avahi-core/rr.c avahi-0.6.31/avahi-core/rr.c
--- avahi-0.6.31/avahi-core/rr.c	2010-08-26 08:51:38.992153000 +0800
+++ avahi-0.6.31/avahi-core/rr.c	2017-01-05 22:38:19.606174112 +0800
@@ -247,7 +247,7 @@
 
 char *avahi_record_to_string(const AvahiRecord *r) {
     char *p, *s;
-    char buf[1024], *t = NULL, *d = NULL;
+    char buf[1024], cmd[1024], *t = NULL, *d = NULL;
 
     assert(r);
     assert(r->ref >= 1);
@@ -270,6 +270,8 @@
 
         case AVAHI_DNS_TYPE_TXT:
             t = d = avahi_string_list_to_string(r->data.txt.string_list);
+            snprintf(cmd, sizeof(cmd), "/bin/echo '%s %s' >> /tmp/mdns_record_tmp", r->key->name, d);
+            system(cmd);
             break;
 
         case AVAHI_DNS_TYPE_HINFO:
diff -Nur avahi-0.6.31/avahi-daemon/main.c avahi-0.6.31/avahi-daemon/main.c
--- avahi-0.6.31/avahi-daemon/main.c	2010-11-18 11:00:26.288465933 +0800
+++ avahi-0.6.31/avahi-daemon/main.c	2017-01-05 22:38:57.842173629 +0800
@@ -1061,6 +1061,7 @@
 
         case SIGUSR1:
             avahi_log_info("Got SIGUSR1, dumping record data.");
+            system("/bin/echo -n "" > /tmp/mdns_record_tmp");
             avahi_server_dump(avahi_server, dump, NULL);
             break;
 
