--- a/pcap-util.c
+++ b/pcap-util.c
@@ -469,14 +469,23 @@
 		break;
 
 	case DLT_USB_LINUX:
+#ifndef PCAP_SUPPORT_USB
+		return;
+#endif
 		swap_linux_usb_header(hdr, data, 0);
 		break;
 
 	case DLT_USB_LINUX_MMAPPED:
+#ifndef PCAP_SUPPORT_USB
+		return;
+#endif
 		swap_linux_usb_header(hdr, data, 1);
 		break;
 
 	case DLT_NFLOG:
+#ifndef PCAP_SUPPORT_NETFILTER
+		return;
+#endif
 		swap_nflog_header(hdr, data);
 		break;
 	}
