--- a/libxtables/xtables.c	2020-11-17 12:11:56.244966828 +0300
+++ b/libxtables/xtables.c	2020-11-17 12:47:24.537006704 +0300
@@ -977,12 +977,6 @@
 	struct xtables_match **pos;
 	bool seen_myself = false;
 
-	if (me->next) {
-		fprintf(stderr, "%s: match \"%s\" already registered\n",
-			xt_params->program_name, me->name);
-		exit(1);
-	}
-
 	if (me->version == NULL) {
 		fprintf(stderr, "%s: match %s<%u> is missing a version\n",
 		        xt_params->program_name, me->name, me->revision);
@@ -1161,12 +1155,6 @@
 	struct xtables_target **pos;
 	bool seen_myself = false;
 
-	if (me->next) {
-		fprintf(stderr, "%s: target \"%s\" already registered\n",
-			xt_params->program_name, me->name);
-		exit(1);
-	}
-
 	if (me->version == NULL) {
 		fprintf(stderr, "%s: target %s<%u> is missing a version\n",
 		        xt_params->program_name, me->name, me->revision);
