--- a/libxtables/xtables.c	2020-11-17 12:47:24.537006704 +0300
+++ b/libxtables/xtables.c	2020-11-17 12:48:55.359964615 +0300
@@ -402,6 +402,7 @@
 
 int xtables_insmod(const char *modname, const char *modprobe, bool quiet)
 {
+#if 0
 	char *buf = NULL;
 	char *argv[4];
 	int status;
@@ -446,6 +447,7 @@
 	free(buf);
 	if (WIFEXITED(status) && WEXITSTATUS(status) == 0)
 		return 0;
+#endif
 	return -1;
 }
 
