--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 # Makefile.am for top level of netatalk package
 
-SUBDIRS = include libatalk bin config etc contrib distrib doc man macros test
+SUBDIRS = include libatalk bin config etc contrib distrib doc man macros
 
 EXTRA_DIST = CONTRIBUTORS COPYRIGHT COPYING Dockerfile NEWS VERSION
 
--- a/configure.ac
+++ b/configure.ac
@@ -319,8 +319,6 @@
   man/man8/cnid_dbd.8
   man/man8/cnid_metad.8
   man/man8/netatalk.8
-  test/Makefile
-  test/afpd/Makefile
 ])
 
 AC_CONFIG_COMMANDS([default],
